### Author :Shanzida Jahan Siddique 
### Date : 06/06/2021
### Titile: QC of PanC4 datasets:
#################################################################

### In general protocol 
##################################################################
# https://www.huble.org/wp-content/uploads/2020/06/HubLE-I-Methods-GWAS-11.pdf

### Related Papers
###############################################################
# https://pubmed.ncbi.nlm.nih.gov/26098869/
# https://pubmed.ncbi.nlm.nih.gov/29422604/
# https://pubmed.ncbi.nlm.nih.gov/26658419/
  
### File	 preparation.	 Create	 binary	 files
#################################################################
# I have got .bed,.bim and .fam file from Alison named CIDR_Klein_PanCa_TOP_subject_level_filtered.bed
# CIDR_Klein_PanCa_TOP_subject_level_filtered.bim,CIDR_Klein_PanCa_TOP_subject_level_filtered.fam
# These files are in my home directory /users/sjahansi.
# In the following code if I include data.ped and data.map and use the command -make bed then I will get three files.bed.bim and .fam.
# PLINK	--file	data	--make-bed	--out	mydata
# This	command	will	create	three	files:	mydata.bed,	mydata.fam	and	mydata.bim.
# To	read	the	binary	file	format,	use	--bfile	mydata.


### Description and purpose of all types of files (https://plink.readthedocs.io/en/latest/plink_fmt/)
###############################################################
#.bed file=The bed file is a binary file containing the genotype information. Must be accompanied by .bim and .fam files.
#.bim file=The bim file contains the SNP names and map positions. Chromosome number, SNP names (i.e., rs ID), and physical positions based on hg19 should be specified. Note that for SNPs with rsID, the rsID should be specified as their SNP names. For SNPs with no rsID, their SNP names should be specified as “chrom:position”. For example, 1:10000 for a SNP at 10000 bp on chromosome 1. 
#.fam file='the fam file contains the family structure
#.ped file= Original standard text format for sample pedigree information and genotype calls. Normally must be accompanied by a .map file; Haploview requires an accompanying .info file instead.

# Contains no header line, and one line per sample with 2V+6 fields where V is the number of variants. The first six fields are the same as those in a .fam file. The seventh and eighth fields are allele calls for the first variant in the .map file ('0' = no call); the 9th and 10th are allele calls for the second variant; and so on.
#.map file=Variant information file accompanying a .ped text pedigree + genotype table. Also generated by "--recode rlist".


#plink.bed      ( binary file, genotype information )
#plink.fam      ( first six columns of mydata.ped ) 
#plink.bim      ( extended MAP file: two extra cols = allele names)

### DEscription of the datasets 
###############################################################

### Methods 
###############################################################

### QC steps
###############################################################

## remove autosome 
./plink --bfile /users/sjahansi/CIDR_Klein_PanCa_TOP_subject_level_filtered --autosome --make-bed --mpheno 2 --pheno /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/PANC4_status.txt --out /
/dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_autosome

## remove geno with the file of remove autosome 
./plink --bfile /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_autosome  --geno 0.01 --make-bed --mpheno 2 --pheno /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/PANC4_status.txt --out /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_geno

## remove mind with the file of remove geno
./plink --bfile /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_geno  --mind 0.02 --make-bed --mpheno 2 --pheno /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/PANC4_status.txt --out /users/sjahansi/remove_mind

## Log file of file_mind

[jhpce01 /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4]$ cat file_mind.log    
PLINK v1.90b6.21 32-bit (19 Oct 2020)
Options in effect:
  --bfile /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_geno
--make-bed
--mind 0.02
--missing
--mpheno 2
--out /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind
--pheno /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/PANC4_status.txt

Hostname: jhpce01.cm.cluster
Working directory: /users/sjahansi/bin/plink
Start time: Tue Jun  1 11:15:45 2021

Random number seed: 1622560545
7969 MB RAM detected; reserving 2047 MB for main workspace.
893594 variants loaded from .bim file.
8094 people (4576 males, 3518 females) loaded from .fam.
8003 phenotype values present after --pheno.
5 people removed due to missing genotype data (--mind).
IDs written to
/dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind.irem .
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 7991 founders and 98 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate in remaining samples is 0.999287.
--missing: Sample missing data report written to
/dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind.imiss, and
variant-based missing data report written to
/dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind.lmiss.
893594 variants and 8089 people pass filters and QC.
Among remaining phenotypes, 4167 are cases and 3831 are controls.  (91
                                                                    phenotypes are missing.)
--make-bed to
/dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind.bed +
  /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind.bim +
  /dcl01/klein/data/imputation/PANC4_Michigan/output/PanC4/file_mind.fam ...
done.

# In QC steps why we used --geno and --mind command 
#############################################################

# Alison said When making the .vcf you need to use both the geno filter command and mind command in plink. geno removes SNPs with missing rates  and mind removes individuals with high missing rates
# She also said In the future, it may make sense to make a bed file that is post initial QC, this way you can just use the bed moving foward and don't have to recreate the QC steps each time.
# In QC steps why I used --geno command first and --mind command last 
#########################################################################

If I used --mind command first and then --geno command then I will get less individuals than expected.
So,by using --geno command at first I created a intermediate bedfile/plink datasets.
So I got less umber of snps this time()
Then with that datasets I created --mind command and got the closer number of individuals.
The audio on this was saved named documents.in gmail.

# remove maf and Hardy-Weinberg equilibrium with the file with remove mind 

#I said Allison that "I have one enquiry.In the Childs paper, in QC steps they used the following criteria : minor allele frequencies (MAF) less than 0.005, Hardy-Weinberg equilibrium p-value<10−6.Would I apply these with the new bed file I recently made with geno 0.01 and mind 0.02 and make a new bed file.Because in the paper they said Overall 654,470 SNPs passed the quality control filters applied and I had got 893594 after geno 0.01 and mind 0.02 command.
# But she said For the childs paper, this is what was included in the analysis (for the association tests). The MAF filtering and HWE filtering can be done AFTER imputation and are dependent on the analysis we are doing.
# Therefore I didn't use here maf and hwe filter or any other filters.
# I used just -autosome --geno 0.01 and mind 0.02 command.

# Association tests 
###############################################################

#QQ plot and Manhattan plot 
###############################################################
#  Next step 
###############################################################

# Need to create .VCF file by following the steps in data preparation of Michigan Imputation server.
# When create .vcf file need to ckeck that the .vcf file has same variants number 893594 and individuals 4167.
# Here are some individuals less than the child paper and Alison said about that the case control number look better now. - a bit more than in the childs paper, which is okay because some get dropped later due to missing covariates. i iwill look at the files you send
# the .imiss files were helpful. In the imiss you can see that all of the genotypes that were set to missing were being used in the calculation of the individual missing rates, there were about 30,000 genotypes set to missing as they had call rates<99%.  You can see in column N_Geno the 925426 genotypes used in the calculations of the individual missing rates- therefore the denominator for the individual missing included all the SNPs we set to missing due to poor call rates. When you made the new bed file these SNPS with the poor call rates were dropped, leaving 893594 SNPs,  The mind.imiss shows then that when you did the mind filter the denominator for the missing was this smaller number 893594,  therefore fewer individuals were dropped.
# confirm that all of these individuals are dropped and the new .vcf has 7998 individuals

